helix-core/src/line_ending.rs

```diff
@@ -203,6 +203,13 @@ pub fn line_end_char_index(slice: &RopeSlice, line: usize) -> usize {
             .unwrap_or(0)
 }
 
+pub fn line_end_byte_index(slice: &RopeSlice, line: usize) -> usize {
+    slice.line_to_byte(line + 1)
```

Comment:
Is there any risk of going over the line count here with the `+ 1`?

Reply:
Hmm this is something I didn't consider and a good point.

I just copied the `line_end_char_idx` function and converted it to return a byte index. For this specific usecase it's fine since ropey allows one past the end index but LSP doesn't (and that's enforced at the callsite of this function at the start of `lsp_pos_to_pos`. We should probably check all existing call sites of `line_end_char_idx` but that would be seprerste from this PR as it's an existing function.